src/Handler/Impl/Lock/SymfonyLockHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenClassrooms\ServiceProxy\Handler\Impl\Lock;
+
+use OpenClassrooms\ServiceProxy\Handler\Contract\LockHandler;
+use OpenClassrooms\ServiceProxy\Handler\Exception\LockException;
+use OpenClassrooms\ServiceProxy\Handler\Impl\ConfigurableHandler;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\LockFactory;
+use Symfony\Component\Lock\LockInterface;
+
+final class SymfonyLockHandler implements LockHandler
+{
+    use ConfigurableHandler;
+
+    /**
+     * @var LockInterface[]
+     */
+    private array $locks = [];
+
+    public function __construct(
+        private readonly LockFactory $lockFactory
+    ) {
+    }
+
+    /**
+     * @throws LockException
+     */
+    public function acquire(string $key): void
+    {
+        try {
+            $this->locks[$key] = $this->lockFactory->createLock($key);
```

Comment:
Should we check if the locks already exists ? 

Without we may have a situation where we deadlock ourself (trying to acquire an already acquired lock in the same process)

Reply:
when testing this was not produceable, but I added the check as it has no downside